src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -7363,32 +7497,73 @@ internal static List<CompletionResult> CompleteHashtableKey(CompletionContext co
+                                {
+                                    return GetSpecialHashTableKeyMembers(excludedKeys, wordToComplete, "LogName", "ProviderName", "Path", "Keywords", "ID", "Level",
+                                    "StartTime", "EndTime", "UserID", "Data");
+                                }
+                                else if (!hashtableIsNested)
+                                {
+                                    return GetSpecialHashTableKeyMembers(excludedKeys, wordToComplete, "LogName", "ProviderName", "Path", "Keywords", "ID", "Level",
                                     "StartTime", "EndTime", "UserID", "Data", "SuppressHashFilter");
+                                }
+                                return null;
                         }
                     }
 
                     if (parameterName.Equals("Arguments", StringComparison.OrdinalIgnoreCase))
                     {
-                        switch (binding.CommandName)
+                        if (!hashtableIsNested)
                         {
-                            case "Invoke-CimMethod":
-                                var result = new List<CompletionResult>();
-                                NativeCompletionCimCommands(parameterName, binding.BoundArguments, result, commandAst, completionContext, excludedKeys, binding.CommandName);
-                                // this method adds a null CompletionResult to the list but we don't want that here.
-                                if (result.Count > 1)
-                                {
-                                    result.RemoveAt(result.Count - 1);
-                                    return result;
-                                }
+                            switch (binding.CommandName)
+                            {
+                                case "Invoke-CimMethod":
+                                    results = new List<CompletionResult>();
+                                    NativeCompletionCimCommands(parameterName, binding.BoundArguments, results, commandAst, completionContext, excludedKeys, binding.CommandName);
+                                    // this method adds a null CompletionResult to the list but we don't want that here.
+                                    if (results.Count > 1)
+                                    {
+                                        results.RemoveAt(results.Count - 1);
+                                        return results;
+                                    }
+                                    return null;
+                            }
+                        }
+                        return null;
+                    }
 
-                                return null;
+                    IEnumerable<PSTypeName> inferredTypes;
+                    if (hashtableIsNested)
```

Comment:
By reading the code, it doesn't seem you are expecting a nested hashtable here. Did you mean to use `if (!hashtableIsNested)` here?